app/src/main/java/org/projectbuendia/client/utils/Utils.java

```diff
@@ -79,36 +84,65 @@
                 return ((String) a).compareTo((String) b);
             }
             return (a == null ? 0 : intA != null ? 1 : 2)
-                - (b == null ? 0 : intB != null ? 1 : 2);
+                    - (b == null ? 0 : intB != null ? 1 : 2);
         }
     };
     /**
      * Compares two lists, each of whose elements is a null, Integer, Long,
      * BigInteger, or String, lexicographically by element, just like Python.
      */
     public static Comparator<List<Object>> nullIntStrListComparator =
-        new Comparator<List<Object>>() {
-            @Override public int compare(List<Object> a, List<Object> b) {
-                for (int i = 0; i < Math.min(a.size(), b.size()); i++) {
-                    int result = nullIntStrComparator.compare(a.get(i), b.get(i));
-                    if (result != 0) {
-                        return result;
+            new Comparator<List<Object>>() {
+                @Override
+                public int compare(List<Object> a, List<Object> b) {
+                    for (int i = 0; i < Math.min(a.size(), b.size()); i++) {
+                        int result = nullIntStrComparator.compare(a.get(i), b.get(i));
+                        if (result != 0) {
+                            return result;
+                        }
                     }
+                    return a.size() - b.size();
                 }
-                return a.size() - b.size();
-            }
-        };
+            };
+
+
+    /**
+     * Sorts an ArrayList of ObsRow rows so that the order of the ObsRow.conceptUids matches the
+     * order of conceptRowUuids.
+     */
+    public static ArrayList<ObsRow> sortObsRows(final ArrayList<ObsRow> rows,
+                                                final ArrayList<String> conceptUuids) {
+
+        Collections.sort(rows, new Comparator<ObsRow>() {
+                    @Override
```

Comment:
This is one level of indentation, so it should be 4 spaces in; it looks like it's 12.